### PR TITLE
WL-4020 Get public announcements for a non accessible site.

### PR DESCRIPTION
**Pull request by ouit0408**
For merged announcements, if user has no access to the site get public
messages from the site.Added method 'getMessagesPublic' to fetch all
public messages from the channel.

### CODE REVIEW

File: announcement-tool/tool/src/java/org/sakaiproject/announcement/entityprovider/AnnouncementEntityProviderImpl.java [hunk 1]
```diff
@@ -224,6 +224,11 @@ public String getEntityPrefix() {
 				announcements.addAll(announcementService.getMessages(channel, t, numberOfAnnouncements, true, false, onlyPublic));
 			} catch (PermissionException e) {
 				log.warn("User: " + currentUserId + " does not have access to view the announcement channel: " + channel + ". Skipping...");
+				//user may not have access to view the channel but get all public messages in this channel
+				AnnouncementChannel announcementChannel = (AnnouncementChannel)announcementService.getChannelPublic(channel);
+				if(announcementChannel != null){
+					announcements.addAll(announcementChannel.getMessagesPublic(null, true));
```
Comment (buckett) [inline announcement-tool/tool/src/java/org/sakaiproject/announcement/entityprovider/AnnouncementEntityProviderImpl.java]: This doesn't respect the time parameter that's passed into the method.

**Comment by ouit0408** [comment]
Added time restriction for public messages.

File: announcement-tool/tool/src/java/org/sakaiproject/announcement/entityprovider/AnnouncementEntityProviderImpl.java [hunk 2]
```diff
@@ -224,6 +225,17 @@ public String getEntityPrefix() {
 				announcements.addAll(announcementService.getMessages(channel, t, numberOfAnnouncements, true, false, onlyPublic));
 			} catch (PermissionException e) {
 				log.warn("User: " + currentUserId + " does not have access to view the announcement channel: " + channel + ". Skipping...");
+				//user may not have access to view the channel but get all public messages in this channel
+				AnnouncementChannel announcementChannel = (AnnouncementChannel)announcementService.getChannelPublic(channel);
+				if(announcementChannel != null){
+					List<Message> publicMessages = announcementChannel.getMessagesPublic(null, true);
+					for(Message message : publicMessages){
+						//Add message only if it is within the time range
+						if(isMessageWithinPastNDays(message, numberOfDaysInThePast)){
```
Comment (buckett) [inline announcement-tool/tool/src/java/org/sakaiproject/announcement/entityprovider/AnnouncementEntityProviderImpl.java:234]: It also doesn't do the number of announcements (`numberOfAnnouncements`).

Comment (ouit0408) [inline announcement-tool/tool/src/java/org/sakaiproject/announcement/entityprovider/AnnouncementEntityProviderImpl.java:234]: as later in the method trimming of decorated announcements for the list is done

Comment (buckett) [inline announcement-tool/tool/src/java/org/sakaiproject/announcement/entityprovider/AnnouncementEntityProviderImpl.java:234]: Ok.

**Pull request merged**